scala/scala.bzl

```diff
@@ -484,37 +479,38 @@ def _scala_binary_impl(ctx):
       rjars = rjars,
       main_class = ctx.attr.main_class,
       jvm_flags = ctx.attr.jvm_flags,
-      args = "",
-      run_before_binary = "",
-      run_after_binary = "",
   )
   return _scala_binary_common(ctx, cjars, rjars)
 
 def _scala_repl_impl(ctx):
   # need scala-compiler for MainGenericRunner below
-  jars = _collect_jars_from_common_ctx(ctx, extra_runtime_deps = [ctx.attr._scalacompiler])
-  (cjars, rjars) = (jars.compiletime, jars.runtime)
-  rjars += [ctx.outputs.jar]
+  jars = _collect_jars_from_common_ctx(
+      ctx, extra_runtime_deps = [ctx.attr._scalacompiler]
+  )
+  cjars = jars.compiletime
+  rjars = jars.runtime + [ctx.outputs.jar]
 
-  args = " ".join(ctx.attr.scalacopts)
   _write_launcher(
       ctx = ctx,
       rjars = rjars,
       main_class = "scala.tools.nsc.MainGenericRunner",
       jvm_flags = ["-Dscala.usejavacp=true"] + ctx.attr.jvm_flags,
-      args = args,
-      run_before_binary = """
+      args = " ".join(ctx.attr.scalacopts),
+      preamble = """
```

Comment:
As discussed, add reminder for future maintainers to 'watch out' for the java launcher using trap as well.